main.tf

```diff
@@ -429,3 +429,11 @@ resource "github_repository_webhook" "repository_webhook" {
     secret       = try(var.webhooks[count.index].secret, null)
   }
 }
+
+resource "github_actions_secret" "repository_secret" {
+  count = length(var.secrets)
```

Comment:
This is fine and works well but has the downside of recreating secrets when elements are added to the list of secrets..

To fix this behavior, I would suggest to use `for_each` here and key it by `secret_name`.

another option would be to not have a list as input type but a map:

**benefits**:
  - no duplicate keys (`secret_name`) by design
  - less code to write as you can just map `secret_name => plaintext_value` 

**downside:**
 - not having `plaintext` in such a clear way.

The downside could be tackled by renaming the input variable to `plaintext_secrets` to make this clear to the user that this input is sensitive and you might not want to check it into your git repository but just grab it from some data source

wdyt?

Reply:
I didn't know about this issue when using `count` to create resources.I'll check the options you mentioned
Thanks!

Reply:
as long as the names are known in the plan phase it makes sense to use them as keys for . it gets problematic once you want to use `for_each` on computed values (depend on resources you create in the same run).. but I guess secret names are static and not the result of creating other resources.. 

when using count it is safe to add new values to the end of a list and removing them from the end. once you remove them in the middle or beginning or add them there, you will see a cascading set of changes being planned and applied.. 

In this case this would be of little impact, I assume, as it would just recreate all secrets.. the chance of hitting a situation where a secret is recreated via terraform while a CI job tries to read it is very very low.. ;)

Reply:
Just updated this resource using `for_each` and a map type instead of a list of objects. Readme and test have been updated accordingly too.

I didn't find any other better name for this variable, so I took your sugestion and rename it to be `plaintext_secrets`